messaging/src/main/java/org/axonframework/messaging/deadletter/InMemoryDeadLetterQueue.java

```diff
@@ -0,0 +1,373 @@
+                       letter.setExpiresAt(expiresAt);
+                       releasedGroups.add(letter.queueIdentifier().group());
+                   });
+
+        releasedGroups.stream()
+                      .map(availabilityCallbacks::get)
+                      .filter(Objects::nonNull)
+                      .forEach(scheduledExecutorService::submit);
+    }
+
+    @Override
+    public void clear(@Nonnull Predicate<QueueIdentifier> queueFilter) {
+        List<QueueIdentifier> queuesToClear = deadLetters.keySet()
+                                                         .stream()
+                                                         .filter(queueFilter)
+                                                         .collect(Collectors.toList());
+
+        queuesToClear.forEach(queueId -> {
+            deadLetters.get(queueId).clear();
+            deadLetters.remove(queueId);
+            logger.info("Cleared out all letters for queue [{}].", queueId);
+        });
+    }
+
+    /**
+     * Builder class to instantiate an {@link InMemoryDeadLetterQueue}.
+     * <p>
+     * The maximum number of queues defaults to {@code 1024}, the maximum amount of dead letters inside a queue defaults
+     * to {@code 1024}, the dead letter expire threshold defaults to a {@link Duration} of 5000 milliseconds, and the
+     * {@link ScheduledExecutorService} defaults to a {@link Executors#newSingleThreadScheduledExecutor(ThreadFactory)},
+     * using an {@link AxonThreadFactory}.
+     *
+     * @param <T> The type of {@link Message} maintained in this {@link DeadLetterQueue}.
+     */
+    public static class Builder<T extends Message<?>> extends SchedulingDeadLetterQueue.Builder<Builder<T>, T> {
+
+        private int maxQueues = 1024;
+        private int maxQueueSize = 1024;
+
+        /**
+         * Sets the maximum number of queues this {@link DeadLetterQueue} may contain. This requirement reflects itself
+         * as the maximum amount of unique {@link QueueIdentifier QueueIdentifiers} referencing a dead-letter queue.
+         * <p>
+         * The given {@code maxQueues} is required to be a positive number, higher or equal to {@code 128}. It defaults
+         * to {@code 1024}.
+         *
+         * @param maxQueues The maximum amount of queues for the queue under construction.
+         * @return The current Builder, for fluent interfacing.
+         */
+        public Builder<T> maxQueues(int maxQueues) {
+            assertThat(maxQueues,
+                       value -> value >= 128,
+                       "The maximum number of queues should be larger or equal to 128");
+            this.maxQueues = maxQueues;
+            return this;
+        }
+
+        /**
+         * Sets the maximum amount of {@link DeadLetter letters} per queue this {@link DeadLetterQueue} can store.
+         * <p>
+         * The given {@code maxQueueSize} is required to be a positive number, higher or equal to {@code 128}. It
```

Comment:
128 seems to be an arbitrary number. Any reason why we wouldn't allow, for example, just 4 items?

Reply:
No reason. This was "natte vinger werk" (translation to English is terrible...). Four is fine by me too, although it seems *very* low. I mean, this dictates the total amount of letters in the queue. 

Zero doesn't make sense, as you can't enqueue anything in that case. Similarly, 1 seems off, as setting up this entire process for a single letter seems off too. I felt similar about any low numbers, honestly.

Reply:
I gave it some additional thought and think having a lowerbound other than 1 doesn't make to much sense. 
I'd rather not make assumption on this, as I can't foresee all scenarios.

As such, the assertion will be adjusted to check for strictly positive numbers on both max-sequences and max-sequence-size.